graphstorm-processing/docs/source/usage/example.rst

```diff
@@ -0,0 +1,249 @@
+  in the configuration file, the corresponding file is expected to exist at ``${PATH}/${FILEPATH}``.
+
+For example:
+
+.. code-block:: bash
+
+    > pwd
+    /home/path/to/data/ # This is the current working directory (cwd)
+    > ls
+    gconstruct-config.json edge_data # These are the files under the cwd
+    > ls edge_data/ # These are the files under the edge_data director
+    movie-included_in-genre.csv
+
+The contents of the ``gconstruct-config.json`` can be:
+
+.. code-block:: json
+
+    {
+    "edges" : [
+        {
+            "files": ["edges/movie-included_in-genre.csv"],
+            "format": {
+                "name": "csv",
+                "separator" : ","
+            }
+        }
+      ]
+    }
+
+Given the above we can run a job with local input data as:
+
+.. code-block:: bash
+
+    > gs-processing --input-data /home/path/to/data \
+        --config-filename gsprocessing-config.json
+
+The benefit with using relative paths is that we can move the same files
+to any location, including S3, and run the same job without making changes to the config
+file:
+
+.. code-block:: bash
+
+    # Move all files to new directory
+    > mv /home/path/to/data /home/new-path/to/data
+    # After moving all the files we can still use the same config
+    > gs-processing --input-data /home/new-path/to/data \
+        --config-filename gsprocessing-config.json
+
+    # Upload data to S3
+    > aws s3 sync /home/new-path/to/data s3://my-bucket/data/
+    # We can still use the same config, just change the prefix to an S3 path
+    > python run_distributed_processing.py --input-data s3://my-bucket/data \
+        --config-filename gsprocessing-config.json
+
+Node files are optional
+^^^^^^^^^^^^^^^^^^^^^^^
+
+GSProcessing does not require node files to be provided for
+every node type. If a node type appears in one of the edges,
+its unique node identifiers will be determined by the edge files.
+
+In the example GConstruct above, the node ids for the node types
+``movie`` and ``genre`` will be extracted from the edge list provided.
+
+Example data and configuration
+------------------------------
+
+For this example we use a small heterogeneous graph inspired by the Movielens dataset.
+You can see the configuration file under
+``graphstorm-processing/tests/resources/small_heterogeneous_graph/gconstruct-config.json``
+
+We have 4 node types, ``movie``, ``genre``, ``director``, and ``user``. The graph has 3
+edge types, ``movie:included_in:genre``, ``user:rated:movie``, and ``director:directed:movie``.
+
+We include one ``no-op`` feature, ``age``, that we directly pass to the output without any transformation,
+and one label, ``gender``, that we transform to prepare the data for a node classification task.
+
+
+Run a GSProcessing job locally
+------------------------------
+
+While GSProcessing is designed to run on distributed clusters on Amazon SageMaker,
```

Comment:
Do we need to explicitly say this is designed for Amazon SageMaker?
I would expect GSProcessing can run on any Spark env.

Reply:
Good point, I'll remove the SageMaker mention.